platforms/core-runtime/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -92,12 +92,18 @@ public void featureUsed(final DeprecatedFeatureUsage usage) {
                 @Override
                 public void execute(InternalProblemSpec builder) {
                     ProblemSpec problemSpec = builder
-                        .label(usage.formattedMessage())
+                        .label(usage.getSummary())
                         .documentedAt(usage.getDocumentationUrl());
                     addPossibleLocation(diagnostics, problemSpec);
                     // TODO (donat) we can further categorize deprecation warnings https://github.com/gradle/gradle/issues/26928
-                    problemSpec.category(DefaultProblemCategory.DEPRECATION, TextUtil.screamingSnakeToKebabCase(usage.getType().name()))
+                    problemSpec.category(DEPRECATION, screamingSnakeToKebabCase(usage.getType().name()))
                         .severity(WARNING);
+                    if(usage.getAdvice() != null) {
+                        problemSpec.solution(usage.getAdvice());
+                    }
+                    if(usage.getContextualAdvice() != null) {
+                        problemSpec.solution(usage.getContextualAdvice());
```

Comment:
🤔 @donat Do we have a contextual solution now as well? It might be good to distinguish the two.

Reply:
A distinction on our side is still not there.
I wanted to deliver what was already here before I messed up the PR from @hegyibalint

Reply:
I'm working on it: #27889